src/credentials/credential-wallet.ts

```diff
@@ -266,101 +266,23 @@ export class CredentialWallet implements ICredentialWallet {
     if (!schema.$metadata.uris['jsonLdContext']) {
       throw new Error('jsonLdContext is missing is the schema');
     }
-    request.context = request.context ?? [];
+    // do copy of request to avoid mutation
+    const r = { ...request };
+    r.context = r.context ?? [];
     if (
-      request.displayMethod?.type === DisplayMethodType.Iden3BasicDisplayMethodV1 &&
-      !request.context.includes(VerifiableConstants.JSONLD_SCHEMA.IDEN3_DISPLAY_METHOD)
+      r.displayMethod?.type === DisplayMethodType.Iden3BasicDisplayMethodV1 &&
+      !r.context.includes(VerifiableConstants.JSONLD_SCHEMA.IDEN3_DISPLAY_METHOD)
     ) {
-      request.context.push(VerifiableConstants.JSONLD_SCHEMA.IDEN3_DISPLAY_METHOD);
+      r.context.push(VerifiableConstants.JSONLD_SCHEMA.IDEN3_DISPLAY_METHOD);
     }
-    const context = [
-      VerifiableConstants.JSONLD_SCHEMA.W3C_CREDENTIAL_2018,
-      ...request.context,
-      VerifiableConstants.JSONLD_SCHEMA.IDEN3_CREDENTIAL,
-      schema.$metadata.uris['jsonLdContext']
-    ];
+    r.context.push(schema.$metadata.uris['jsonLdContext']);
+    r.expiration = !r.expiration || r.expiration == 0 ? undefined : r.expiration * 1000;
+    r.id = r.id ? r.id : `urn:${uuid.v4()}`;
+    r.issuanceDate = r.issuanceDate ? r.issuanceDate * 1000 : Date.now();
 
-    const credentialType = [
-      VerifiableConstants.CREDENTIAL_TYPE.W3C_VERIFIABLE_CREDENTIAL,
-      request.type
-    ];
-
-    const expirationDate =
-      !request.expiration || request.expiration == 0 ? null : request.expiration;
-
-    const credentialSubject = request.credentialSubject;
-    credentialSubject['type'] = request.type;
-
-    const cr = new W3CCredential();
-    cr.id = `urn:${uuid.v4()}`;
-    cr['@context'] = context;
-    cr.type = credentialType;
-    cr.expirationDate = expirationDate ? new Date(expirationDate * 1000).toISOString() : undefined;
-    cr.refreshService = request.refreshService;
-    cr.displayMethod = request.displayMethod;
-    cr.issuanceDate = new Date().toISOString();
-    cr.credentialSubject = credentialSubject;
-    cr.issuer = issuer.string();
-    cr.credentialSchema = {
-      id: request.credentialSchema,
-      type: VerifiableConstants.JSON_SCHEMA_VALIDATOR
-    };
-
-    cr.credentialStatus = this.buildCredentialStatus(request, issuer);
-
-    return cr;
+    return new W3CCredential().fromCredentialRequest(issuer, r);
```

Comment:
Please change signature of function to `W3CCredential.fromCredentialRequest(...)`

Reply:
done. Did the method static